### PR TITLE
grouping of commands in help

### PR DESCRIPTION
**Pull request by maxandersen**
Implemented the grouping but not all sure about the grouping - feedback welcome:

```
Usage: jbang [-h] [--verbose] [COMMAND]
Compiles and runs .java/.jsh scripts.
  -h, --help      Display help/info
      --verbose   jbang will be verbose on what it does.

Basic Commands:
  run         Compile and run provided .java/.jsh script.
  edit        Edit a .java/.jsh script.
  init        Initialize a .java/.jsh script.

Advanced Commands:
  build       Build .java/.jsh script without running. Use for caching
                dependencies.
  trust       Manage trust domains.
  cache       Cache management.

Catalog:
  alias       Manage aliases.
  catalog     Manage catalogs.

Other:
  version     Display version info.
  completion  Output auto-completion script for bash/zsh.
              Usage: source <(jbang completion)

Copyright: 2020 Max Rydahl Andersen, License: MIT
Website: https://github.com/jbangdev/jbang
```

### CODE REVIEW

**Comment by quintesse** [comment]
- Maybe reword "Compile and run provided .java/.jsh script." to "Run provided .java/.jsh script." as suggested by @abelsromero
? The fact that it compiles is perhaps not interesting for a user.
 - Perhaps reword "Build .java/.jsh script without running. Use for caching dependencies.", especially the "Use for caching dependencies" is somewhat vague and perhaps overly technical. Perhaps "Can be used to speed up a future run"?

For the rest the grouping seems like a very good idea.

**Comment by abelsromero** [comment]
> * Maybe reword "Compile and run provided .java/.jsh script." to "Run provided .java/.jsh script." as suggested by @abelsromero
>   ? The fact that it compiles is perhaps not interesting for a user.
> 
> * Perhaps reword "Build .java/.jsh script without running. Use for caching dependencies.", especially the "Use for caching dependencies" is somewhat vague and perhaps overly technical. Perhaps "Can be used to speed up a future run"?

As [mentioned in the discussion](https://github.com/jbangdev/jbang/discussions/269) if the idea is not to hide certain details from users, I think we can use direct references to the commands so the users know what's going on. For example describing `run` using the term `build` points the reader to the `build` command to better understand the whole workflow.

```
run    Builds and runs...

build  Compiles and stores .java/.jsh script in the cache without running.

cache Manage compiled scripts in the local cache.
```

> Implemented the grouping

Not sure about the grouping, the division of basic and advanced seems arbitrary. I'd just group all them together order of most used to least maybe: run, build, cache, init, edit, trust

**Comment by maxandersen** [comment]
Thanks @quintesse and @abelsromero - did a update based on your comments; and it now looks like this:

```
Usage: jbang [-h] [--verbose] [COMMAND]
Build and runs .java/.jsh scripts and jar packages.
  -h, --help      Display help/info
      --verbose   jbang will be verbose on what it does.

Essentials:
  run         Builds and runs provided script or jar package.
  build       Compiles and stores script in the cache.
  cache       Manage compiled scripts in the local cache.

Editing:
  init        Initialize a script.
  edit        Setup a temporary project to edit script in an IDE.

Configuration:
  trust       Manage which domains you trust to run scripts from.
  alias       Manage aliases for scripts.
  catalog     Manage Catalogs of aliases.
  jdk         Manage Java Development Kits installed by jbang.

Other:
  completion  Output auto-completion script for bash/zsh.
              Usage: source <(jbang completion)
  version     Display version info.

Copyright: 2020 Max Rydahl Andersen, License: MIT
Website: https://jbang.dev
```
I'm including a screenshot as the visual ansi coding does help a bit:




A few things I did in latest tweaks:

- used the idea of using commands in the description "*build*s and *run* ..."
- kept groups as I find its a bit more readable with the extra "air", but this time a bit more descriptive/relevant groups
- removed repetitive .java/.jsh and just use script or jar package where relevant
- unhide `jdk` command
- update website to be jbang.dev and not github page
- (not shown) when using --verbose and error happens in exception handler will point users to https://github.com/jbangdev/jbang/issues.

wdyt ?

**Comment by quintesse** [comment]
I like having groups and I like the newer descriptions. I'm not sure about the groups though, Some if it seems a bit arbitrary. Still, it's not as if its important to get things right the first time, we can slowly improve this over time.

But to give some comments:
 - `cache` being in essentials seems strange to me, I basically never use the command :-)
 - and putting `trust`, `alias`, `catalog` and `jdk` together in `Configuration` seems a bit weird, them all being so different and mostly having nothing to do with configuration :-)
 - I feel that `cache` and `jdk` could be put together, both having something to do with the management of downloads, but not sure what the group could be called.
 - And perhaps `trust` could go to `Essentials` seeing that without it you can't run anything remote
 - And that would leave `alias` and `catalog` which are very closely related. BUt no idea what a good group name would be.

Not very helpful perhaps if I can't come up with good names, sorry ;-)

**Comment by abelsromero** [comment]
It seems fine for me.

I also checked a few examples and found:
* Tools that do NOT group:  go, cargo, rustc, nvm, npm, sdk, java
* Tools that group: kubectl, rvm

In both grouping cases there are som arbitrary classifications, and kubectl does use "basic" and "advanced" groups which to me seem "uninspired" to say the least :laughing: So, maybe just release and see how ppl reacts.

PS: Shoudn't the first `Usage` line exemplify a correct execution instead of how to obtain help?

**Comment by maxandersen** [comment]
okey, round 3, changes this time around:

- Much more details and upfront listing of usage examples 
- added jbang contributors to copyright
- reordered groups ...

```
jbang is a tool for building and running .java/.jsh scripts and jar packages.
Usage: jbang [-h] [--verbose] [COMMAND]

  jbang init hello.java [args...]
        (to initialize a script)
  or  jbang edit --live=code hello.java
        (to edit a script in IDE with live updates)
  or  jbang hello.java [args...]
        (to run a .java file)
  or  jbang gavsearch@jbangdev [args...]
        (to run a alias from a catalog)
  or  jbang group-id:artifact-id:version [args...]
        (to run a .jar file found with a GAV id)

  -h, --help      Display help/info
      --verbose   jbang will be verbose on what it does.

Essentials:
  run         Builds and runs provided script.
  build       Compiles and stores script in the cache.

Editing:
  init        Initialize a script.
  edit        Setup a temporary project to edit script in an IDE.

Caching:
  cache       Manage compiled scripts in the local cache.
  jdk         Manage Java Development Kits installed by jbang.

Configuration:
  trust       Manage which domains you trust to run scripts from.
  alias       Manage aliases for scripts.
  catalog     Manage Catalogs of aliases.

Other:
  completion  Output auto-completion script for bash/zsh.
              Usage: source <(jbang completion)
  version     Display version info.

Copyright: 2020 Max Rydahl Andersen and jbang.dev contributors, License: MIT
Website: https://jbang.dev
```

**Comment by maxandersen** [comment]
decided to merge it in - seemed mainly ok based on other feedback thus lets get it tried out and see if it helps or fails ;)

**Comment by quintesse** [comment]
Hah, I justed wanted to add a suggestion about the `run` command :-)

I wondered if we should perhaps mention that in many cases the word `run` is optional? (Just in case people wonder why there is a `run` command but it's not being mentioned in the usage examples)

**Comment by maxandersen** [comment]
Yeah. Makes sense. Wanna do a pr?

**Comment by quintesse** [comment]
What shall we put? "Builds and runs provided script. The word 'run' is optional in most cases."?
